programs/just.nix

```diff
@@ -1,40 +1,32 @@
 {
   lib,
-  pkgs,
-  config,
   mkFormatterModule,
   ...
 }:
 let
-  cfg = config.programs.just;
+  # Search 'justfile' in subdirectories too. This isn't needed for filename starting with `*.`.
+  alsoSearchSubdirectory = lib.lists.concatMap (fileName: [ fileName ] ++ [ "*/${fileName}" ]);
 in
 {
   meta.maintainers = [ "katexochen" ];
 
   imports = [
     (mkFormatterModule {
       name = "just";
-      includes = [
-        "[Jj][Uu][Ss][Tt][Ff][Ii][Ll][Ee]" # 'justfile', case insensitive
-        ".[Jj][Uu][Ss][Tt][Ff][Ii][Ll][Ee]" # '.justfile', case insensitive
-        "*.[Jj][Uu][Ss][Tt]" # '.just' module, case insensitive
-        "*.[Jj][Uu][Ss][Tt][Ff][Ii][Ll][Ee]" # '.justfile' module, case insensitive
+      args = [
+        "--fmt"
+        "--unstable"
+        "--justfile"
```

Comment:
This simplification is wrong. just itself doesn't comply with the treefmt spec, as the `--justfile` flags expects a single argument, but the spec requires the formatter to accept multiple file names as arguments (see https://treefmt.com/latest/reference/formatter-spec/#1-files-passed-as-arguments). That's the reason for the bash loop (I should have added a comment, sorry for that).

Reply:
Thanks, I didn't notice that at all. I re-added the bash loop :)